cros_gralloc/cros_gralloc_driver.cc

```diff
@@ -109,6 +114,86 @@ cros_gralloc_driver *cros_gralloc_driver::get_instance()
         close(fd);                   \
     }
 
+int32_t cros_gralloc_driver::reload()
+{
+	int fd;
+	drmVersionPtr version;
+	char const *str = "%s/renderD%d";
+	uint32_t j;
+	char *node;
+
+	const int render_num = 10;
+	const int name_length = 50;
+	int availabe_node = 0;
+	uint64_t gpu_grp_type = 0;
+
+	availabe_node = 0;
+	for (uint32_t i = drv_num_; i < drv_num_ + IVSH_DEVICE_NUM; i++) {
+		if (asprintf(&node, str, DRM_DIR_NAME, i) < 0)
+			continue;
+
+		fd = open(node, O_RDWR, 0);
+		free(node);
+		if (fd < 0)
+			continue;
+
+		version = drmGetVersion(fd);
+		if (!version) {
+			close(fd);
+			continue;
+		}
+
+		availabe_node++;
+		drv_num_ = i;
+		close(fd);
+		drmFreeVersion(version);
+	}
+
+	// have the ivshemem node
+	if ((availabe_node > 0) && !drv_ivshmem_) {
+		if (asprintf(&node, str, DRM_DIR_NAME, drv_num_) < 0) {
+			drv_loge("Open new added node fail\n");
+			return -ENODEV;
+		}
+
+		gpu_grp_type = GPU_TYPE_NORMAL;
```

Comment:
This assignment here still will overlap previous GPU_TYPE_DUAL_IGPU_DGPU flag. Use "|=" maybe better?